tests/integration/test_dimension.py

```diff
@@ -47,7 +58,206 @@ def type_fixture(self, request):
+    """Integration-test suite for `cr.cube.dimension.NewDimension` object.
+
+    This is a temporary object, roughly a shim, to provide the new Dimension interface
+    to FrozenSlice. Once that work is completed, this will become DescribeDimension and
+    the NewDimension object will be merged into the Dimension object.
+    """
+
+    def it_constructs_from_a_legacy_Dimension_object(self, legacy_dimension):
+        dimension = NewDimension(legacy_dimension, None)
+        assert isinstance(dimension, NewDimension)
+
+    def it_provides_access_to_all_elements_in_its_collection(self, legacy_dimension):
+        legacy_dimension = CrunchCube(CR.ECON_BLAME_WITH_HS).dimensions[0]
+        dimension_transforms_dict = {}
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        elements = dimension.all_elements
+
+        assert isinstance(elements, _NewAllElements)
+
+    def it_knows_its_transformed_description(self, legacy_dimension):
+        """Including that it resolves the dimension-rename transform cascade."""
+        dimension_transforms_dict = {"description": "foobar"}
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        description = dimension.description
+
+        assert description == "foobar"
+
+    def but_it_uses_element_description_when_not_transformed(self, legacy_dimension):
+        """Including that it resolves the dimension-rename transform cascade."""
+        dimension_transforms_dict = {}
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        description = dimension.description
+
+        assert description == (
+            "If President Obama and the Republicans in Congress do not reach a budget"
+            " agreement in time to avoid a shutdown of the federal government, who do"
+            " you think will more to blame--President Obama or the Republican Congres"
+            "s?"
+        )
+
+    def it_knows_the_display_order_of_its_elements(self, legacy_dimension):
+        dimension_transforms_dict = {
+            "order": {"type": "explicit", "element_ids": [3, 5, 42, 8, 2]}
+        }
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        display_order = dimension.display_order
+
+        assert display_order == (2, 4, 5, 1, 0, 3, 6, 7)
+
+    def it_knows_its_transformed_name(self, legacy_dimension):
+        dimension_transforms_dict = {"name": "barfoo"}
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        name = dimension.name
+
+        assert name == "barfoo"
+
+    def but_it_uses_element_name_when_not_transformed(self, legacy_dimension):
+        dimension_transforms_dict = {}
+        dimension = NewDimension(legacy_dimension, dimension_transforms_dict)
+
+        name = dimension.name
+
+        assert name == "ShutdownBlame"
+
+    def it_knows_whether_it_should_be_pruned(self, legacy_dimension):
```

Comment:
Should we add `but_it_defaults_to_false_if_not_specified`?

Reply:
We could, but I only added these tests to drive the development of NewDimension and defend it until we got it integrated integrated into a single new Dimension object. At that point, I think most of these tests would be replaced by unit tests, which would be much more thorough, like cover all possible execution paths and input combinations. So I didn't try to be exhaustive here.